docs/apache-airflow/best-practices.rst

```diff
@@ -124,10 +124,63 @@ Airflow parses the Python file. For more information, see: :ref:`managing_variab
 Top level Python Code
 ---------------------
 
-In general, you should not write any code outside of defining Airflow constructs like Operators. The code outside the
-tasks runs every time Airflow parses an eligible python file, which happens at the minimum frequency of
+You should avoid writing the top level code which is not necessary to create Operators
+and build DAG relations between them. Specifically you should not run any database access,
+heavy computations and networking operations. The code outside the Operator's ``execute`` methods
+runs every time Airflow parses an eligible python file, which happens at the minimum frequency of
 :ref:`min_file_process_interval<config:scheduler__min_file_process_interval>` seconds.
```

Comment:
Oh yeah. The minimum/maximum notion of time/frequency is reversed I will update it to still keep minimum and refer to time rather than frequency to avoid confusion :D